src/odemis/driver/smaract.py

```diff
@@ -1254,7 +1254,10 @@ def __init__(self, name, role, locator, axes, ref_on_init=False, linear_speed=0.
             logging.debug("SA_MC is referenced")
         else:
             if ref_on_init:
-                self.reference().result()
+                try:
+                    self.reference().result()
+                except Exception as e:
+                    self.state._set_value(e, force_write=True)
```

Comment:
Ok. Will this cause problems though? Because any operations after init will fail if the device is not referenced.

Reply:
The code in _updatePosition() seems to already handle this case properly.
Well, actually, looking at that code, I'd suggest to return an *empty* dict, instead of a dict with all positions at 0. It's clearer that there is no data, instead of pretending we know that it's at 0 (although it's probably wrong).